src/streaming-client.ts

```diff
@@ -179,8 +179,9 @@ export class RevAiStreamingClient extends EventEmitter {
     }
 
     private closeStreams(): void {
-        if (this.streamsClosed === false) {
+        if (!this.streamsClosed) {
             this.streamsClosed = true;
+            this.protocol.end();
```

Comment:
This shouldn't be necessary. According to the node 17 docs:
```
duplex.allowHalfOpen (https://nodejs.org/api/stream.html#duplexallowhalfopen)
If false then the stream will automatically end the writable side when the readable side ends. Set initially by the allowHalfOpen constructor option, which defaults to false.
```

Reply:
I guess I don't see this in the node 14 docs. Is it necessary for that version?

Reply:
I think the `requests` and `responses` Passthroughs is separate from the duplex readable and writable streams? Several unit-tests started failing because the assertion `expect(res.writable).toBe(false)` was failing (where `res` is the returned duplex)